core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2461,7 +2476,7 @@ private[spark] class DAGScheduler(
       clearCacheLocs()
     }
     if (fileLost) {
-      val remove = if (ignoreShuffleFileLostEpoch) {
+      val remove = if (ignoreShuffleFileLostEpoch || isShuffleMerger) {
```

Comment:
`removeExecutorAndUnregisterOutputs` is being called from 2 places:
1. Fetch failure handling: So here `ignoreShuffleFileLostEpoch` is going to be `true`. Then it doesn't matter if the execId is a shuffleMergerId or not.
2. Executor Lost handling: When it is called from here, then it is never going to be a `shuffleMergerId`. 
Given this, I think it might be better to remove this `isShuffleMerger` check here and just add a comment that when the fetch failure is for a merged shuffle chunk, `ignoreShuffleFileLostEpoch` is true, and so all the files will be removed.
WDYT? cc. @mridulm

Reply:
yeah, makes sense. I've removed the check and just added the comment.